src/middlewared/middlewared/plugins/crypto.py

```diff
@@ -1083,12 +1404,12 @@ def __ca_sign_csr(self, data, schema_name):
             'signedby': ca_data['id']
         }
 
-        new_csr_dict = self.middleware.call_sync(
+        new_csr_job_id = self.middleware.call_sync(
```

Comment:
As far as I cam see certificate.create returns a Job now, not an id. Why not wait for it to finish?